ResourceManager/GitExtensionsFormBase.cs

```diff
@@ -126,5 +137,12 @@ protected void TranslateItem(string itemName, object item)
         }
 
         #endregion
+
+        /// <summary>
+        /// Notifies whenever the application becomes active.
+        /// </summary>
+        protected virtual void OnApplicationActivated()
```

Comment:
Why not as an event?

Reply:
Override is better.  Events are not deterministic on which event handler handles the event first. Multiple handlers can be attached.  We can raise an event but this is the right way to do this.

Reply:
> Why not as an event?

Events are significantly more expensive - they involve several new types and classes, you also need to worry about wiring to and unwiring from handlers.
Events are useful when you have no control over the underlying code - like `Form` events. For the code you have control over overrides are much better solution. 
Virtual calls are slightly more expensive than non-virtual calls, but nowhere near a tax you pay with events.